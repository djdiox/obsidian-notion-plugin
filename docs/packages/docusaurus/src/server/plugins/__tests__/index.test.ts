/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import {loadPlugins} from '..';
import type {Plugin, Props} from '@docusaurus/types';

describe('loadPlugins', () => {
  it('loads plugins', async () => {
    const siteDir = path.join(__dirname, '__fixtures__/site-with-plugin');
    await expect(
      loadPlugins({
        siteDir,
        generatedFilesDir: path.join(siteDir, '.docusaurus'),
        outDir: path.join(siteDir, 'build'),
        siteConfig: {
          baseUrl: '/',
          trailingSlash: true,
          themeConfig: {},
          presets: [],
          plugins: [
            () =>
              ({
                name: 'test1',
                prop: 'a',
                async loadContent() {
                  // Testing that plugin lifecycle is bound to the instance
                  return this.prop;
                },
                async contentLoaded({content, actions}) {
                  actions.addRoute({
                    path: 'foo',
                    component: 'Comp',
                    modules: {content: 'path'},
                    context: {content: 'path'},
                  });
                  actions.setGlobalData({content, prop: this.prop});
                },
              } as Plugin & ThisType<{prop: 'a'}>),
          ],
          themes: [
            () => ({
              name: 'test2',
              configureWebpack() {
                return {};
              },
            }),
          ],
        },
        siteConfigPath: path.join(siteDir, 'docusaurus.config.js'),
      } as unknown as Props),
    ).resolves.toMatchSnapshot();
  });
});
