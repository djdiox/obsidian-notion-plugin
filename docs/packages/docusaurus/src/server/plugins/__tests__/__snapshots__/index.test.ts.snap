// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadPlugins loads plugins 1`] = `
{
  "globalData": {
    "test1": {
      "default": {
        "content": "a",
        "prop": "a",
      },
    },
  },
  "plugins": [
    {
      "content": "a",
      "contentLoaded": [Function],
      "loadContent": [Function],
      "name": "test1",
      "options": {
        "id": "default",
      },
      "path": "<PROJECT_ROOT>/packages/docusaurus/src/server/plugins/__tests__/__fixtures__/site-with-plugin",
      "prop": "a",
      "version": {
        "type": "local",
      },
    },
    {
      "configureWebpack": [Function],
      "content": undefined,
      "name": "test2",
      "options": {
        "id": "default",
      },
      "path": "<PROJECT_ROOT>/packages/docusaurus/src/server/plugins/__tests__/__fixtures__/site-with-plugin",
      "version": {
        "type": "local",
      },
    },
    {
      "content": undefined,
      "getClientModules": [Function],
      "injectHtmlTags": [Function],
      "name": "docusaurus-bootstrap-plugin",
      "options": {
        "id": "default",
      },
      "path": "<PROJECT_ROOT>/packages/docusaurus/src/server/plugins/__tests__/__fixtures__/site-with-plugin",
      "version": {
        "type": "synthetic",
      },
    },
    {
      "configureWebpack": [Function],
      "content": undefined,
      "name": "docusaurus-mdx-fallback-plugin",
      "options": {
        "id": "default",
      },
      "path": ".",
      "version": {
        "type": "synthetic",
      },
    },
  ],
  "pluginsRouteConfigs": [
    {
      "component": "Comp",
      "context": {
        "data": {
          "content": "path",
        },
        "plugin": "<PROJECT_ROOT>/packages/docusaurus/src/server/plugins/__tests__/__fixtures__/site-with-plugin/.docusaurus/test1/default/plugin-route-context-module-100.json",
      },
      "modules": {
        "content": "path",
      },
      "path": "foo/",
    },
  ],
}
`;
